yamux/src/frame.rs

```diff
@@ -13,89 +13,144 @@ mod io;
 
 use futures::future::Either;
 use header::{Data, GoAway, Header, Ping, StreamId, WindowUpdate};
-use std::{convert::TryInto, num::TryFromIntError};
+use std::{convert::TryInto, fmt::Debug, marker::PhantomData, num::TryFromIntError};
+use zerocopy::{AsBytes, ByteSlice, ByteSliceMut, Ref};
 
 pub use io::FrameDecodeError;
 pub(crate) use io::Io;
 
-/// A Yamux message frame consisting of header and body.
-#[derive(Clone, Debug, PartialEq, Eq)]
+use self::header::HEADER_SIZE;
+
+/// A Yamux message frame consisting of header and body in a single buffer
+#[derive(Clone, Debug, Eq, PartialEq)]
 pub struct Frame<T> {
-    header: Header<T>,
-    body: Vec<u8>,
+    buffer: Vec<u8>,
+    _marker: std::marker::PhantomData<T>,
+}
+
+impl<T> Default for Frame<T> {
```

Comment:
I don't think this should exist. Doesn't this more or less circumvent all the type safety around `_marker` if one can instantiation a `Frame` of any `T` just like that?

Reply:
True. I previously added Default in order to call std::mem::take on a Frame in io.rs. Now I have changed std::mem::take with std::mem::replace.